File: test-src/20_propertyGetterSetter.kt
class FunkyClass {
  var internalString = ""

  var wrappedProperty: String
    get() { return "My string is $internalString" }
    set(value) {
      internalString = "$value - previous=\"$internalString\""
    }
}

val computedProperty1: Int
  get() {
    var a = 0
    a++
    return a
  }

val computedProperty2: Int
  get() { return 2 }

/*val computedProperty3: Int
  get() = 3*/


var _backingProperty: Int = 0
var computedProperty4: Int
  get() {
      return 4 + _backingProperty
  }
  set(value) {
      _backingProperty = value
  }

var computedProperty5: Int
  get() { return 5 + _backingProperty }
  set(value) { _backingProperty = value }

/*var computedProperty6: Int
  get() = 6 + _backingProperty
  set(value) { _backingProperty = value }*/

fun main(args: Array<String>) {
  val x = FunkyClass()

  println(x.wrappedProperty)
  x.wrappedProperty = "abc"
  println(x.wrappedProperty)
  x.wrappedProperty = "123"
  println(x.wrappedProperty)

  println(computedProperty1)
  println(computedProperty2)
  //println(computedProperty3)

  println(computedProperty4)
  println(computedProperty5)
  //println(computedProperty6)
  computedProperty4 = 1000
  println(computedProperty4)
  println(computedProperty5)
  //println(computedProperty6)
}

File: test-dest/20_propertyGetterSetter.swift
public class FunkyClass {
  var internalString = ""

  var wrappedProperty: String {
    get { return "My string is \(internalString)" }
    set(value) {
      internalString = "\(value) - previous=\"\(internalString)\""
    }
}
  init() {
  }
}

var computedProperty1: Int32 {
  get {
    var a = 0
    a++
    return a
  }
}

var computedProperty2: Int32 {
  get { return 2 }
}

/*val computedProperty3: Int
  get() = 3*/


var _backingProperty: Int32 = 0
var computedProperty4: Int32 {
  get {
      return 4 + _backingProperty
  }
  set(value) {
      _backingProperty = value
  }
}

var computedProperty5: Int32 {
  get { return 5 + _backingProperty }
  set(value) { _backingProperty = value }
}

/*var computedProperty6: Int
  get() = 6 + _backingProperty
  set(value) { _backingProperty = value }*/

func main(args: [String]) {
  let x = FunkyClass()

  print(x.wrappedProperty)
  x.wrappedProperty = "abc"
  print(x.wrappedProperty)
  x.wrappedProperty = "123"
  print(x.wrappedProperty)

  print(computedProperty1)
  print(computedProperty2)
  //print(computedProperty3)

  print(computedProperty4)
  print(computedProperty5)
  //print(computedProperty6)
  computedProperty4 = 1000
  print(computedProperty4)
  print(computedProperty5)
  //print(computedProperty6)
}

main([])
